lib/1.4/dml-builtins.dml

```diff
@@ -407,6 +407,52 @@ template post_init {
     param _each_post_init = each post_init in (this);
 }
 
+/**
+### destroy
+
+Provides an abstract method `destroy`, which is called when the
+device is being deleted. Typically used to `delete` any device state that has
+been dynamically allocated using `new`.
+
+The method `destroy` is automatically called on all objects that implement the
+`destroy` template. The order in which `destroy` of objects are called is not
+defined, except that `destroy` of a particular object is guaranteed to be
+called before `destroy` of any of its parent objects.
+In particular, `destroy` of the device object will be called only after all
+other implementations of `destroy`.
+
+The `destroy` template can't be instantiated for `event` objects. This is
+because `event` objects, through the `custom_time_event` or
+`custom_cycle_event` template, may already require defining a method named
+`destroy` whose purpose is different from the one required by the `destroy`
+template. To work around this limitation, you may declare a `group` within the
+`event` object that instantiates the `destroy` template instead.
```

Comment:
This section maybe deserves to be deemphasised through `<div class="note"> **Note:**`

Reply:
Yup, and if so, it should be moved to the last part of the docs.